File: src/graph/model/CollapseChange.java
package graph.model;

import graph.model.IGraphModel.AtomicGraphModelChange;

public class CollapseChange extends AtomicGraphModelChange
{

	/**
	 *
	 */
	protected Object cell;

	/**
	 * 
	 */
	protected boolean collapsed, previous;

	/**
	 * 
	 */
	public CollapseChange()
	{
		this(null, null, false);
	}

	/**
	 * 
	 */
	public CollapseChange(GraphModel model, Object cell,
			boolean collapsed)
	{
		super(model);
		this.cell = cell;
		this.collapsed = collapsed;
		this.previous = this.collapsed;
	}

	/**
	 * 
	 */
	public void setCell(Object value)
	{
		cell = value;
	}

	/**
	 * @return the cell
	 */
	public Object getCell()
	{
		return cell;
	}

	/**
	 * 
	 */
	public void setCollapsed(boolean value)
	{
		collapsed = value;
	}

	/**
	 * @return the collapsed
	 */
	public boolean isCollapsed()
	{
		return collapsed;
	}

	/**
	 * 
	 */
	public void setPrevious(boolean value)
	{
		previous = value;
	}

	/**
	 * @return the previous
	 */
	public boolean getPrevious()
	{
		return previous;
	}

	/**
	 * Changes the root of the model.
	 */
	public void execute()
	{
		collapsed = previous;
		previous = ((GraphModel) model)._collapsedStateForCellChanged(
				cell, previous);
	}

}

File: lib/model/collapse_change.dart
part of graph.model;

class CollapseChange extends AtomicGraphModelChange {

  Object cell;

  bool collapsed, previous;

  CollapseChange([GraphModel model = null, Object cell = null, bool collapsed = false]) : super(model) {
    this.cell = cell;
    this.collapsed = collapsed;
    this.previous = this.collapsed;
  }

  void setCell(Object value) {
    cell = value;
  }

  /**
   * Returns the cell.
   */
  Object getCell() {
    return cell;
  }

  void setCollapsed(bool value) {
    collapsed = value;
  }

  bool isCollapsed() {
    return collapsed;
  }

  void setPrevious(bool value) {
    previous = value;
  }

  /**
   * Returns the previous.
   */
  bool getPrevious() {
    return previous;
  }

  /**
   * Changes the root of the model.
   */
  void execute() {
    collapsed = previous;
    previous = (model as GraphModel)._collapsedStateForCellChanged(cell, previous);
  }

}
